src/pgduckdb_ddl.cpp

```diff
@@ -371,16 +371,30 @@ DuckdbHandleDDL(PlannedStmt *pstmt, const char *query_string, ParamListInfo para
 		return;
 	} else if (IsA(parsetree, RenameStmt)) {
 		auto stmt = castNode(RenameStmt, parsetree);
-		if (stmt->renameType != OBJECT_SCHEMA) {
-			/* We only care about schema renames for now */
-			return;
-		}
-		if (strncmp("ddb$", stmt->subname, 4) == 0) {
-			elog(ERROR, "Changing the name of a ddb$ schema is currently not supported");
+		if (stmt->renameType == OBJECT_SCHEMA) {
+			if (strncmp("ddb$", stmt->subname, 4) == 0) {
+				elog(ERROR, "Changing the name of a ddb$ schema is currently not supported");
+			}
+			if (strncmp("ddb$", stmt->newname, 4) == 0) {
+				elog(ERROR, "Changing a schema to a ddb$ schema is currently not supported");
+			}
 		}
-		if (strncmp("ddb$", stmt->newname, 4) == 0) {
-			elog(ERROR, "Changing a schema to a ddb$ schema is currently not supported");
+		if (stmt->renameType == OBJECT_TABLE || stmt->renameType == OBJECT_COLUMN) {
+			Oid relation_oid = RangeVarGetRelid(stmt->relation, AccessShareLock, false);
+			Relation relation = RelationIdGetRelation(relation_oid);
```

Comment:
Interestingly `relation` is colored as a keyword by GH... Not sure if that's a big deal, but we could name it `rel` instead too without loosing readability?